File: Python/Medium/Sorting/451__Sort_Characters_By_Frequency.py
from collections import Counter

# 1 - method

# def frequencySort(s: str) -> str:
#     char = {}
#     for i in s:
#         char[i] = char.get(i, 0) + 1
#
#     result = ""
#     for i, j in sorted(char.items(), key=lambda x: x[1], reverse=True):
#         result += j * i
#
#     return result

# 2 method
def frequencySort(s: str) -> str:
    count = Counter(s)
    result = ''.join(char * freq for char, freq in count.most_common())
    return result



s = 'Aabb'
print(frequencySort(s))

File: PHP/Medium/Sorting/451__Sort_Characters_By_Frequency.php
<?php
function frequencySort($s) {
    $charCount = array_count_values(str_split($s));
    arsort($charCount);

    $result = '';
    foreach ($charCount as $char => $count) {
        $result .= str_repeat($char, $count);
    }
    return $result;
}

echo frequencySort("tree");
//echo frequencySort("Aabb");